src/os_maild/sendmail.c

```diff
@@ -505,49 +503,52 @@ int OS_Sendmail(MailConfig *mail, struct tm *p)
         }
     }
 
-    if (mail->to) {
-        /* Add CCs */
-        if (mail->to[1]) {
-            i = 1;
-            while (1) {
-                if (mail->to[i] == NULL) {
-                    break;
-                }
-
-                memset(snd_msg, '\0', 256);
-                snprintf(snd_msg, 256, TO, mail->to[i]);
-
-                if (sendmail) {
-                    fprintf(sendmail->file_in, "%s", snd_msg);
-                } else {
-                    OS_SendTCP(socket, snd_msg);
-                }
-
-                i++;
+    if (mail->to || mail->gran_to) {
+        char to_list[1024];
+        char *p      = to_list;
+        size_t rem   = sizeof(to_list);
+        bool first   = true;
+
+        /* Macro to add each mailbox between <> and separated by “, ” */
+        #define APPEND(addr) do {                                  \
+        if (!first) {                                            \
+            int n = snprintf(p, rem, ", ");                       \
+            if (n < 0 || (size_t)n >= rem) break;                 \
+            p += n; rem -= n;                                      \
+        }                                                        \
+        int n2 = snprintf(p, rem, "<%s>", addr);                 \
+        if (n2 < 0 || (size_t)n2 >= rem) break;                  \
+        p += n2; rem -= n2;                                      \
+        first = false;                                           \
+        } while(0)
+
+        /* main */
+        if (mail->to) {
+        for (int i = 0; mail->to[i]; i++) {
+            APPEND(mail->to[i]);
+        }
+        }
+        /* granular */
+        if (mail->gran_to) {
+        for (int i = 0; mail->gran_to[i]; i++) {
+            if (mail->gran_set[i] == FULL_FORMAT) {
+            APPEND(mail->gran_to[i]);
             }
         }
-    }
-
-    /* More CCs - from granular options */
-    if (mail->gran_to) {
-        i = 0;
-        while (mail->gran_to[i] != NULL) {
-            if (mail->gran_set[i] != FULL_FORMAT) {
-                i++;
-                continue;
-            }
-
-            memset(snd_msg, '\0', 256);
-            snprintf(snd_msg, 256, TO, mail->gran_to[i]);
+        }
+        #undef APPEND
 
-            if (sendmail) {
-                fprintf(sendmail->file_in, "%s", snd_msg);
-            } else {
-                OS_SendTCP(socket, snd_msg);
+        /* Send single TO: if there is at least one */
+        if (p != to_list) {
+        if (sendmail) {
+            fprintf(sendmail->file_in, TO, to_list);
+        } else {
+            char hdr[1080];
+            int hl = snprintf(hdr, sizeof(hdr), TO, to_list);
+            if (hl > 0 && (size_t)hl < sizeof(hdr)) {
+            OS_SendTCP(socket, hdr);
             }
-
-            i++;
-            continue;
+        }
```

Comment:
can we keep the tabs inside each group?
like this:
```c
        if (p != to_list) {
            if (sendmail) {
                fprintf(sendmail->file_in, TO, to_list);
            } else {
                char hdr[1080];
                int hl = snprintf(hdr, sizeof(hdr), TO, to_list);
                if (hl > 0 && (size_t)hl < sizeof(hdr)) {
                    OS_SendTCP(socket, hdr);
                }
            }
```

there are several cases in the same file